Assets/Scripts/StandardSamples/Services/EOSService.cs

```diff
@@ -67,6 +72,63 @@ protected EOSService(bool requiresAuthentication = true)
             _ = RefreshAsync();
         }
 
+        /// <summary>
+        /// Dispose function makes sure that the manager is removed from the
+        /// collection of connect login listeners.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+
+            // Prevent collection until the dispose pattern is followed.
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Protected implementation of Dispose pattern. If this method is
+        /// overridden in implementing classes, this base implementation should
+        /// be called at the end of the overridden implementation.
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
+        {
+            // If already disposed, then do nothing (this prevents the disposal
+            // pattern from becoming circular).
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                // NOTE: Dispose of managed resources here.
+                Reset();
+            }
+
+            // NOTE: Free unmanaged resources here, and set large fields to null.
+
```

Comment:
This is where the event for `AuthenticationListener` should be unsubscribed from.